File: packages/navigation/ios/RNRNavigationContainerManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNRNavigationContainerManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onWillShowView, RCTDirectEventBlock)

RCT_EXTERN_METHOD(
            pop:(nonnull NSNumber *)node
        animated:(nonnull BOOL *)animated
)

RCT_EXTERN_METHOD(
            popTo:(nonnull NSNumber *)node
        viewNode:(nonnull NSNumber *)viewNode
        animated:(nonnull BOOL *)animated
)

@end

File: packages/navigation/ios/RNRNavigationContainerManager.swift
@objc(RNRNavigationContainerManager)
class RNRNavigationContainerManager: RCTViewManager {
    var navigationContainer: RNRNavigationContainer?

    override func view() -> UIView! {
        navigationContainer = RNRNavigationContainer()
        return navigationContainer
    }

    @objc func pop(_ node: NSNumber, animated: Bool) {
        DispatchQueue.main.async { [self] in
            (bridge.uiManager.view(
                    forReactTag: node
            ) as? RNRNavigationContainer)?.pop(animated: animated)
        }
    }

    @objc func popTo(_ node: NSNumber, viewNode: NSNumber, animated: Bool) {
        DispatchQueue.main.async { [self] in
            let view = (bridge.uiManager.view(
                    forReactTag: viewNode
            ) as? RNRNavigationScene)
            if view != nil {
                (bridge.uiManager.view(
                        forReactTag: node
                ) as? RNRNavigationContainer)?.popTo(view!, animated: animated)
            }
        }
    }

    override static func requiresMainQueueSetup() -> Bool {
        true
    }
}
